src/library-aux/scala/Any.scala

```diff
@@ -138,15 +138,107 @@ abstract class Any {
+   * [scala.util.Failure], throw an Exception etc.)
+   *
+   * Following some examples targeting the JVM and JDK8 using Scala 3.4.0. We start off with statements where the
+   * outcome is most likely as you would expect and continue with more bizarre outcomes -- hoping that you really 
+   * refrain from using it as type testing utility, use [isInstanceOf] instead.
+   *
+   * {{{
+   * // doesn't insert any operation 1 is statically known to be a subtype of Int
+   * 1.asInstanceOf[Int]
+   * // inserts a narrowing primitive conversion from double to int
+   * 1.0.asInstanceOf[Int]
+   * // inserts a widening primitive conversion from int to long (which could fail with an OutOfMemoryError)
+   * 1.asInstanceOf[Long]
+   * // inserts a widening primitive conversion and boxing
+   * List[Int](1.0.asInstanceOf[Int])
+   * // inserts boxing and a checkcast to java.lang.Double, this fails as no conversion takes place
+   * 1.asInstanceOf[java.lang.Double]
+   *
+   * // doesn't insert any operation as B is statically know to be a subtype of A
+   * B().asInstanceOf[A]
+   *
+   * // doesn't insert any operation, A is statically known to be a subtype of A
+   * (B(): A).asInstanceOf[A]
+   *
+   * // inserts a checkcast to B
+   * (B(): A).asInstanceOf[B]
+   *
+   * // inserts boxing but no checkcast because List[Int] and List[String] have both the type `List` after type erasure
+   * // and as we have seen before, if it is already known that a subtype relation holds, no checkcast is necessary
+   * (1 :: Nil).asInstanceOf[List[String]]
+   *
+   * // inserts a checkcast for X which fails
+   * // as said, no guarantee that this stays this way. 
+   * // also checkcast to java.lang.Object or no operation at all would be have been valid solutions
+   * "a".asInstanceOf[ Y | Z ]
+   *
+   * // doesn't insert any operation, outputs "a"
+   * println("a".asInstanceOf[ Y | Int ])
+   *
+   * // boxes and inserts a checkcast to java.lang.Object
+   * 1.asInstanceOf [ Y | Float ]
+   *
+   * // doesn't inserts any operation
+   * 1.asInstanceOf[ String & Int ]
+   *
+   * // inserts an unboxingToInt which uses a checkcast to java.lang.Integer internally which fails
+   * "a".asInstanceOf[ String & Int ]
+   *
+   * // inserts an unboxingToInt where 0 is used as default value
+   * println(null.asInstanceOf[Int])
+   *
+   * // inserts an unboxingToInt where 0 is used as default value, prints out 0
+   * println(null.asInstanceOf[2])
+   *
+   * // inserts an unboxingToInt where 0 is used as default value 
+   * val two: 2 = null.asInstanceOf[2] 
+   * // value of two is ignored and the compiler computes a constant expression (2 + 1) instead
+   * // i.e. the result is still 3
+   * println(two + 1)
+   *
+   * val twoFromJava: java.lang.Integer = null
+   * // doesn't insert any operation, outputs `null`
+   * println(twoFromJava.asInstanceOf[2])
+   *
+   * // doesn't insert any operation
+   * val one = 2.asInstanceOf[1]
+   *
+   * }}}
```

Comment:
Maintaining "2 space gutter" (two spaces after asterisk) would not be amiss.

"The default implementation just throws an exception, so anything the compiler may do to improve upon this outcome may be considered a win." (Or maybe don't say that.)

Reply:
I guess this was a victim of the auto-format, will do